docs/source/java/modules.rst

```diff
@@ -0,0 +1,413 @@
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: java
+.. highlight:: java
+
+.. _java_modules:
+
+=======
+Modules
+=======
+
+.. contents::
+
+This is the variety of arrow java documentation:
+
+* `Specification and protocols`_: This contains agnostic specification that is implemented in this case by arrow java modules.
+* `Supported environment`_ (like this): This contains answers for what-is-that arrow java module.
+* `Cookbook`_: This contains answers about how-to-use arrow java modules with practices examples.
+* `Development`_: This contains detailed information about what you need to consider to start with arrow java development.
+
+Arrow java modules is created using specification such as columnar format, off-heap
+memory, serialization and interprocess communication (IPC). Some of the java modules
+was created with their own native implementations and others through bindings.
+
+.. list-table:: Arrow Java Implementations and Bindings
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Module
+     - Root Decision
+   * - Arrow Memory
+     - Native implementation.
+   * - Arrow Vector
+     - Native implementation.
+   * - Arrow Flight Grpc
+     - Native implementation.
+   * - Arrow Flight Sql
+     - Native implementation.
+   * - Arrow Algorithm
+     - Native implementation.
+   * - Arrow Compression
+     - Native implementation.
+   * - Arrow C Data Interface
+     - Native implementation.
+   * - Arrow Dataset
+     - Bindings
+   * - Arrow ORC
+     - Bindings
+   * - Arrow Gandiva
+     - Bindings
+
+Arrow java is divided in these modules to offer in-memory columnar data structures:
+
+.. list-table:: Arrow Java Modules
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Module
+     - Description
+   * - arrow-format
+     - Generated Java files from the IPC Flatbuffer definitions.
+   * - arrow-memory-core
+     - Core off-heap memory management libraries for Arrow ValueVectors.
+   * - arrow-memory-unsafe
+     - Allocator and utils for allocating memory in Arrow based on sun.misc.Unsafe.
+   * - arrow-memory-netty
+     - Netty allocator and utils for allocating memory in Arrow.
+   * - arrow-vector
+     - An off-heap reference implementation for Arrow columnar data format.
+   * - arrow-tools
+     - Java applications for working with Arrow ValueVectors.
+   * - arrow-jdbc
+     - (`Experimental`) A library for converting JDBC data to Arrow data.
```

Comment:
We don't need backticks around 'experimental', and we can consolidate these two tables (if you decide to keep them as tables)